Cannot print Tables with added metadata

In the TestTableCollectionPickle we call ``self.add_metadata`` to all the input tables before testing. However, if we try to ``print()`` those tables we get an error. Inserting a ``print(tables)`` at line 3887 gives

```
tests/test_tables.py:3887: in verify
    print(tables)
tskit/tables.py:2482: in __str__
    str(self.nodes),
tskit/tables.py:455: in __str__
    headers, rows = self._text_header_and_rows(
tskit/tables.py:882: in _text_header_and_rows
    util.truncate_string_end(str(row.metadata)),
tskit/metadata.py:714: in __get__
    row, "_metadata", row._metadata_decoder(row._metadata)
tskit/metadata.py:566: in <lambda>
    self.decode = lambda buffer: decoder(iter(buffer))
tskit/metadata.py:357: in <lambda>
    return lambda buffer: {
tskit/metadata.py:358: in <dictcomp>
    key: sub_decoder(buffer) for key, sub_decoder in sub_decoders.items()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

buffer = <bytes_iterator object at 0x7fed522fa4f0>

>   return lambda buffer: struct.unpack(f, bytes(islice(buffer, size)))[
        0
    ].decode(encoding)
E   struct.error: unpack requires a buffer of 50 bytes

tskit/metadata.py:390: error
```


(As an aside, while I was looking through the metadata code trying to figure out what was going on, I was confused by the use of ``buffer`` as a variable name, as it's a Python built-in, which could easily be called in this context. ``buff`` or something would probably be a better choice)

--- a/python/tests/test_tables.py
+++ b/python/tests/test_tables.py
@@ -3076,13 +3076,15 @@ class TestTableCollection:
         for table in tskit.TABLE_NAMES:
             t = getattr(tc, table)
             if hasattr(t, "metadata_schema"):
-                t.packset_metadata([f"{table}-{i}".encode() for i in range(t.num_rows)])
+                t.packset_metadata(
+                    [f"{table}-{i:10}".encode() for i in range(t.num_rows)]
+                )
                 t.metadata_schema = tskit.MetadataSchema(
                     {
                         "codec": "struct",
                         "type": "object",
                         "properties": {
-                            table: {"type": "string", "binaryFormat": "50p"}
+                            table: {"type": "string", "binaryFormat": "16p"}
                         },
                     }
                 )

